### PR TITLE
[AC-9535] Include revised program interest in Personal Profile progress calculation

### PR DESCRIPTION
**Pull request by AmosWels**
## [AC-9535](https://masschallenge.atlassian.net/browse/AC-9535)

**Changes introduced**
- add `program_interest` field as part of the  get profile completion progress computation

**Testing**
- Confirm that the definition of done according to [AC-9535](https://masschallenge.atlassian.net/browse/AC-9635) is satisfied

[Sibling PR](https://github.com/masschallenge/accelerate/pull/3268)

### CODE REVIEW

File: accelerator/models/core_profile.py [hunk 1]
```diff
@@ -407,6 +411,7 @@ def is_judge(self, program=None, state=None):
 
     def percent_complete(self):
         completed_count = 0
+        self.add_fields()
```
Comment (iankigen) [inline accelerator/models/core_profile.py:414]: this implementation of this function is not correct. simplified correct implementation:
```...
profile_fields = PROFILE_FIELDS.copy()
if flag_smith_has_feature(PROGRAM_INTEREST_FLAG):
    profile_fields += ['program_interest']
else:
   profile_fields += ['program_families']
...
# replace `PROFILE_FIELDS` references with `profile_fields`

Comment (iankigen) [inline accelerator/models/core_profile.py:414]: this implementation of this function is not correct. simplified correct implementation:
```...
profile_fields = PROFILE_FIELDS.copy()
if flag_smith_has_feature(PROGRAM_INTEREST_FLAG):
    profile_fields += ['program_interest']
else:
   profile_fields += ['program_families']
...
# replace `PROFILE_FIELDS` references with `profile_fields`

File: accelerator/tests/test_core_profile.py [hunk 2]
```diff
@@ -408,11 +411,31 @@ def test_completion_percentage_is_correct_for_completed_profile(self):
...
+
+    @patch("bullet_train.BulletTrain.feature_enabled", return_value=False)
+    @patch("bullet_train.BulletTrain.has_feature", return_value=False)
+    def test_completion_percentage_is_correct_for_incomplete_profile(self,
+                                                                     *args):
         # missing 7/22 fields used to calculate profile completion %
         profile = ExpertProfileFactory(
             program_families=[ProgramFamilyFactory()])
-        self.assertEqual(profile.percent_complete(), 0.68)
+        self.assertEqual(profile.percent_complete(), 0.67)
```
Comment (iankigen) [inline accelerator/tests/test_core_profile.py:438]: percent_complete should be 0.68 when the feature flag is off assuming the implementation is correct

Comment (iankigen) [inline accelerator/tests/test_core_profile.py:438]: percent_complete should be 0.68 when the feature flag is off assuming the implementation is correct

**Review by iankigen — APPROVED** [review]

**Review by kenware — APPROVED** [review]
Looks good

**Pull request merged**